sdks/python/apache_beam/yaml/yaml_ml.py

```diff
@@ -33,11 +40,419 @@
+        return python_callable.PythonCallableWithSource(callable)
+      else:
+        raise ValueError(
+            f"Must specify one of 'callable' or 'path' and 'name' for {typ} "
+            f"function.")
+
+    if processing_transform:
+      if isinstance(processing_transform, dict):
+        return _parse_config(**processing_transform)
+      else:
+        raise ValueError("Invalid model_handler specification.")
+
+  def underlying_handler(self):
+    return self._handler
+
+  def preprocess_fn(self, row):
+    raise ValueError(
+        'Handler does not implement a default preprocess '
+        'method. Please define a preprocessing method using the '
+        '\'preprocess\' tag.')
+
+  def create_preprocess_fn(self):
+    return lambda row: (row, self._preprocess(row))
+
+  @staticmethod
+  def postprocess_fn(x):
+    return x
+
+  def create_postprocess_fn(self):
+    return lambda result: (result[0], self._postprocess(result[1]))
+
+  @staticmethod
+  def validate(model_handler_spec):
+    raise NotImplementedError(type(ModelHandlerProvider))
+
+  @classmethod
+  def register_handler_type(cls, type_name):
+    def apply(constructor):
+      cls.handler_types[type_name] = constructor
+      return constructor
+
+    return apply
+
+  @classmethod
+  def create_handler(cls, model_handler_spec) -> "ModelHandlerProvider":
+    typ = model_handler_spec['type']
+    config = model_handler_spec['config']
+    try:
+      result = cls.handler_types[typ](**config)
+      if not hasattr(result, 'to_json'):
+        result.to_json = lambda: model_handler_spec
+      return result
+    except Exception as exn:
+      raise ValueError(
+          f'Unable to instantiate model handler of type {typ}. {exn}')
+
+
+@ModelHandlerProvider.register_handler_type('VertexAIModelHandlerJSON')
+class VertexAIModelHandlerJSONProvider(ModelHandlerProvider):
+  def __init__(
+      self,
+      endpoint_id: str,
+      endpoint_project: str,
+      endpoint_region: str,
+      experiment: Optional[str] = None,
+      network: Optional[str] = None,
+      private: bool = False,
+      min_batch_size: Optional[int] = None,
+      max_batch_size: Optional[int] = None,
+      max_batch_duration_secs: Optional[int] = None,
+      env_vars=None,
+      preprocess: Callable = None,
+      postprocess: Callable = None):
+    """ModelHandler for Vertex AI.
+
+    For example: ::
+
+        - type: RunInference
+          config:
+            inference_tag: 'my_inference'
+            model_handler:
+              type: VertexAIModelHandlerJSON
+              config:
+                endpoint_id: 9876543210
+                endpoint_project: 1234567890
+                endpoint_region: us-east1
+                preprocess:
+                  callable: 'lambda x: {"prompt": x.prompt, "max_tokens": 50}'
+
+    Args:
+      endpoint_id: the numerical ID of the Vertex AI endpoint to query.
+      endpoint_project: the GCP project name where the endpoint is deployed.
```

Comment:
is this a project name or a project number? (Generally we've been using names in Beam.)

Reply:
Maybe we could accept both (and do the name -> number mapping as a follow-up).

Reply:
I thought it only accepted project number according to docs, but it worked with project name. So either works in this case.

I refactored the name to `project` as well as refactored `endpoint_region` to `location` to mimic the underlying ModelHandler